lib/logstash/outputs/udp.rb

```diff
@@ -21,12 +21,19 @@ class LogStash::Outputs::UDP < LogStash::Outputs::Base
   def register
     @socket = UDPSocket.new
     @codec.on_event do |event, payload|
-      @socket.send(payload, 0, @host, @port)
+      begin
+        @socket.send(payload, 0, @host, @port)
+      rescue Errno::EMSGSIZE => e
+        logger.error("Failed to send event, message size of #{payload.size} too long", :error => e.inspect,
+                     :backtrace => e.backtrace.first(10))
+      rescue => e
+        logger.error("Failed to send event:", :error => e.inspect,
```

Comment:
it may be helpful to output at least part of the event in question to the logs if we are effectively dropping it.

Reply:
In that case I'd suggest having a conditional to log the event if only in debug log level